shipitscript/shipitscript/pushlog_scan.py

```diff
@@ -0,0 +1,98 @@
+from enum import Enum, auto, unique
+
+import requests
+
+URL = 'https://hg.mozilla.org/{repo}/json-pushes'
+
+
+@unique
+class Importance(Enum):
+    MAYBE = auto()
+    IMPORTANT = auto()
+    UNIMPORTANT = auto()
+
+
+_push_checks = []
+
+
+def push_check(func):
+    _push_checks.append(func)
+    return func
+
+
+def should_build(repo, old_rev, curr_rev):
+    # TODO redo
+    resp = requests.get(
+        URL.format(repo=repo),
+        params={'full': '1', 'fromchange': old_rev, 'tochange': curr_rev},
+    )
+    pushlog = resp.json()
+    reversed_pushes = [pushlog[k] for k in sorted(pushlog.keys(), reverse=True)]
+    for push in reversed_pushes:
+        push_importance = is_push_important(push)
+        if push_importance == Importance.IMPORTANT:
+            # Tell caller that we want to build.
+            return True
+    return False
+
+
+def is_push_important(push):
+    """
+    Run through all `@push_check`s to find if this `push` is important or not.
+
+    See the help on `Importance` for more information.
+    """
+    isimportant = Importance.MAYBE
+    for check in _push_checks:
+        isimportant = check(push)
+        if not isimportant == Importance.MAYBE:
+            return isimportant
+    # We reached the end of our checks and
+    # we did not explicitly detect important
+    # nor unimportant.
+    # ...Our default is important
+    return Importance.IMPORTANT
```

Comment:
should we log when we end up with a `Importance.MAYBE`?

also, should we initialize `isimportant` as Importance.IMPORTANT above if that's the intended default?

Reply:
addressed them both.